modules/EvseManager/IECStateMachine.cpp

```diff
@@ -0,0 +1,457 @@
+        }
+
+        if (last_cp_state == RawCPState::E || last_cp_state == RawCPState::F) {
+            // Triggers SLAC start
+            events.push(CPEvent::EFtoBCD);
+        }
+        break;
+
+    case RawCPState::D:
+        // If state D is not supported, signal an error event here and switch off.
+        if (not has_ventilation) {
+            call_allow_power_on_bsp(false);
+            events.push(CPEvent::ErrorVentilationNotAvailable);
+            timeout_state_c1.stop();
+            break;
+        }
+        // no break, intended fall through: If we support state D it is handled the same way as state C
+    case RawCPState::C:
+        // Table A.6: Sequence 1.2 Plug-in
+        if (last_cp_state == RawCPState::A) {
+            events.push(CPEvent::CarPluggedIn);
+            EVLOG_info << "Detected simplified mode.";
+            ev_simplified_mode = true;
+        }
+        if (last_cp_state == RawCPState::B) {
+            events.push(CPEvent::CarRequestedPower);
+        }
+
+        if (!pwm_running && last_pwm_running) { // X2->C1
+                                                // Table A.6 Sequence 10.2: EV does not stop drawing power
+                                                // even if PWM stops. Stop within 6 seconds (E.g. Kona1!)
+            timeout_state_c1.start(std::chrono::seconds(6));
+        }
+
+        if (timeout_state_c1.reached()) {
+            EVLOG_warning << "Timeout of 6 seconds reached, EV did not go back to state B after PWM was switch off. "
+                             "Power off under load.";
+            // We are still in state C, but the 6 seconds timeout has been reached. Now force power off under load.
+            call_allow_power_on_bsp(false);
+        }
+
+        if (pwm_running) { // C2
+            // 1) When we come from state B: switch on if we are allowed to
+            // 2) When we are in C2 for a while now and finally get a delayed power_on_allowed: also switch on
+            if (power_on_allowed && (!last_power_on_allowed || last_cp_state == RawCPState::B)) {
+                // Table A.6: Sequence 4 EV ready to charge.
+                // Must enable power within 3 seconds.
+                call_allow_power_on_bsp(true);
+            }
+
+            // Simulate Request power Event here for simplified mode
+            // to ensure that this mode behaves similar for higher
+            // layers. Note this does not work with 5% mode
+            // correctly, but simplified mode does not support HLC
+            // anyway.
+            if (!last_pwm_running && ev_simplified_mode) {
+                events.push(CPEvent::CarRequestedPower);
+            }
+        }
+        break;
+
+    case RawCPState::E:
+        if (last_cp_state != RawCPState::E) {
+            events.push(CPEvent::ErrorE);
+            timeout_state_c1.stop();
+            call_allow_power_on_bsp(false);
+        }
+        break;
+
+    case RawCPState::F:
```

Comment:
I do not really understand why we cannot tell if the car is connected, we see that from state A?
Yes, in the current type definition the BSP does not send any events such as CarUnplugged/requested power etc anymore, that is one of the goals of this PR. They are only used internally in the EvseManager module, and this may go away any time as it does not touch any external interfaces.

Reply:
Lock unlock is not factored out into a separate interface and evse_manager will call lock/unlock commands